packages/app/src/cli/commands/app/scaffold/extension.ts

```diff
@@ -1,222 +1,20 @@
-  validateExtensionTypeLimit(app: AppInterface, type: string | undefined) {
-    if (type && this.limitedExtensionsAlreadyScaffolded(app).includes(type)) {
-      throw new error.Abort(
-        'Invalid extension type',
-        `You can only scaffold one extension of type ${mapExtensionTypeToExternalExtensionType(
-          type as ExtensionTypes,
-        )} per app`,
-      )
-    }
-  }
-
-  validateExtensionFlavor(type: string | undefined, flavor: string | undefined) {
-    if (!flavor || !type) {
-      return
-    }
-    const uiExtensionTemplateNames = getUIExtensionTemplates(type).map((template) => template.value)
-    const functionExtensionTemplateNames = functionExtensionTemplates.map((template) => template.value)
-
-    const invalidTemplateError = (templates: string[]) => {
-      return new error.Abort(
-        'Specified extension template on invalid extension type',
-        `You can only specify a template for these extension types: ${templates.join(', ')}.`,
-      )
-    }
-    if (isUiExtensionType(type) && !uiExtensionTemplateNames.includes(flavor)) {
-      throw invalidTemplateError(uiExtensionTemplateNames)
-    }
-    if (isFunctionExtensionType(type) && !functionExtensionTemplateNames.includes(flavor)) {
-      throw invalidTemplateError(functionExtensionTemplateNames)
-    }
-  }
-
-  /**
-   * Some extension types like `theme` and `product_subscription` are limited to one per app
-   * Use this method to retrieve a list of the limited types that have already been scaffolded
-   *
-   * @param app {AppInterface} current App
-   * @returns {string[]} list of extensions that are limited by quantity and are already scaffolded
-   */
-  limitedExtensionsAlreadyScaffolded(app: AppInterface): string[] {
-    const themeTypes = app.extensions.theme.map((ext) => ext.configuration.type)
-    const uiTypes = app.extensions.ui.map((ext) => ext.configuration.type)
-
-    const themeExtensions = themeTypes.filter((type) => limitedExtensions.theme.includes(type))
-    const uiExtensions = uiTypes.filter((type) => limitedExtensions.ui.includes(type))
-    return [...themeExtensions, ...uiExtensions]
-  }
-
-  formatSuccessfulRunMessage(
-    extensionType: ExtensionTypes,
-    extensionDirectory: string,
-    depndencyManager: PackageManager,
-  ): string {
-    const extensionOutputConfig = getExtensionOutputConfig(extensionType)
-    output.completed(`Your ${extensionOutputConfig.humanKey} extension was added to your project!`)
-
-    const outputTokens = []
-    outputTokens.push(
-      output.content`\n  To find your extension, remember to ${output.token.genericShellCommand(
-        output.content`cd ${output.token.path(extensionDirectory)}`,
-      )}`.value,
+    output.warn(
+      output.content`The command ${output.token.genericShellCommand(
+        `scaffold`,
+      )} has been deprecated in favor of ${output.token.genericShellCommand(
+        `generate`,
+      )} and will be deleted in the next major version.
```

Comment:
In 4.x? If we are not still sure about when, maybe we can just be less explicit for now:

```suggestion
      )} and will be eventually deleted.
```

Reply:
We won't have a 4.0 for quite some time.